### ISSUE TITLE
the recommended value of dim/alpha?

### ISSUE THREAD

**Issue by dherorunner** [KohakuBlueleaf/LyCORIS#29]
I noticed that there are actually two sets of values linear and conv,so what is the the recommended value of them?
Which does dim<64 alpha=1 refer to?

**Comment by KohakuBlueleaf** [comment]
@dherorunner the rank of loha is dim**2
and basically the largest dim in whole SD is 4096, which means if you want to get full rank(4096 rank), you only need 64dim.
And since these things (LoRA/LoHa) are for parameter-efficient finetuning, basically all of them are not "bigger=better".

For LoRA, you should consider about dim=rank=1~16
For LoHA you should consider about dim=sqrt(rank)=2~8

And for alpha, the equation is:
lora_out = lora(x) * alpha/dim
So alpha can be any number.
Basically we will use 1~dim, and if you always use same alpha, you can ignore other hyperparam when you want to adjust the size of dim. (In NN, bigger model need lower lr, but fixed alpha can perform like lr scaler)

So I personally reommend alpha=0.5~1

And I will use lr=1e-4~5e-4 for this kind of setting (with AdamW, weight_decay=0.05~0.2, beta2=0.95~0.99)

**Comment by dherorunner** [comment]
@KohakuBlueleaf 
so the value of network dim and conv dim are the same?

**Comment by KohakuBlueleaf** [comment]
@dherorunner they can be different
